os/arch/arm/src/s5j/s5j_ticksuppress.c

```diff
@@ -0,0 +1,273 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <debug.h>
+#include <queue.h>
+#include <tinyara/wdog.h>
+#include <tinyara/clock.h>
+#include <tinyara/time.h>
+#include <tinyara/wdog.h>
+#include <arch/board/board.h>
+
+#include "up_arch.h"
+#include "chip/s5jt200_rtc.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/*
+ * The desired timer interrupt frequency is provided by the definition
+ * CLK_TCK (see include/time.h).  CLK_TCK defines the desired number of
+ * system clock ticks per second.  That value is a user configurable setting
+ * that defaults to 100 (100 ticks per second = 10 MS interval).
+ *
+ * The timer counts at the rate SYSCLK_FREQUENCY as defined in the board.h
+ * header file.
+ */
+#define SYSTICK_RELOAD ((SYSCLK_FREQUENCY / CLK_TCK) - 1)
+
+#define MAX_TCNT UINT32_MAX	/* Max 32-bit timer-tick value */
+
+/****************************************************************************
+ * Type Definitions
+ ****************************************************************************/
+enum {
+	TICKSUPPRESS_DISABLE,
+	TICKSUPPRESS_ENABLE
+} timertick_mode;
+
+enum {
+	START,
+	READ,
+	WRITE,
+	STOP
+} timer_states;
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+static uint8_t g_tick_mode;	/* timer-tick mode  */
+
+static uint32_t last_event = 0;
```

Comment:
Am not sure, but re-writing these by changing the declaring order would save some bytes ?
static uint32_t last_event = 0;
static uint8_t g_tick_mode;

Reply:
I'm also not sure, could you please explain how?